packages/core/src/bus/EventBus.ts

```diff
@@ -0,0 +1,19 @@
+import { CommandDispatcher } from './CommandDispatcher';
+import { Event } from './Event';
+import { EventDispatcher } from './EventDispatcher';
+import { EventHandler } from './EventHandler';
+
+export interface EventBus extends EventDispatcher, CommandDispatcher {
+  register<T extends Event<R>, R>(
```

Comment:
`EventHandler` should accept a payload instead of an instance of `Event`
```suggestion
  register<T, R>(
```